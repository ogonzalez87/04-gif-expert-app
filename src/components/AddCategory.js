import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    console.log(e.target.value);
    setInputValue(e.target.value);
    console.log("HandleInputChange llamado");
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue("");
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <p> {inputValue} </p>

      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
