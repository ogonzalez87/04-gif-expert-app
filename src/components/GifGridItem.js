import React from "react";
import PropTypes from "prop-types";

<<<<<<< HEAD
=======
 
  

>>>>>>> f08c3ee30ceb030fc5ed63dac3c9031de4155439
export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />

      <p> {title} </p>
    </div>
  );
};
<<<<<<< HEAD
GifGridItem.prototypes = {
=======

GifGridItem.propTypes = {
>>>>>>> f08c3ee30ceb030fc5ed63dac3c9031de4155439
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
