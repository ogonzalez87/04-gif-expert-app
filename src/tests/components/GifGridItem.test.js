import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("<GifGridItem/>", () => {
  test("should load commponent", () => {
    const wrapper = shallow(<GifGridItem />);

    expect(wrapper).toMatchSnapshot();
  });
});
