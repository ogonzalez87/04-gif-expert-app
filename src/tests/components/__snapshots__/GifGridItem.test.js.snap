// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem/> should load commponent 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img />
  <p>
     
     
  </p>
</div>
`;
